src/blockchain.js

```diff
@@ -1,9 +1,51 @@
 /*global ethereum*/
 import {reduce} from './async';
 
-let Blockchain = {};
+let Blockchain = {
+  list: [],
+  done: false,
+  err: null
+};
 let contracts = [];
 
+Blockchain.connect = function({dappConnection, dappAutoEnable = true, warnAboutMetamask, blockchainClient = ''}, cb = () => {}) {
```

Comment:
Tiny nitpick `blockchainClient` could be `null` instead of an empty string. 

Also `warnAboutMetamask`, should probably default to `false`?

And last but not least, considering that this is the first time we're making this API consumer-facing, I think there's no need to support callbacks at this point. I'd vote for making this promise-based only.

Reply:
> blockchainClient could be null

I put empty string in case we have string checks on it, but I know we don't

> Also warnAboutMetamask, should probably default to false

That's a bit useless, because if it's not there, it's gonna be undefined, so falsy anyway.

> I'd vote for making this promise-based only.

I like having both, but we can check with the rest of the team to see what they think